CRM/Extension/System.php

```diff
@@ -286,6 +286,15 @@ public function getRepositoryUrl() {
     return $this->_repoUrl;
   }
 
+  /**
+   * Returns a list keyed by extension key
+   *
+   * @return array
+   */
+  public static function getCompatibilityInfo() {
+    return json_decode(file_get_contents(Civi::paths()->getPath('[civicrm.root]/extension-compatibility.json')), TRUE);
```

Comment:
If I'm reading correctly... every time we build the list of extensions, it'll need to read each `info.xml` individually via `CRM_Extension_Info::parse()` which will then call `filterRequirements()` and  `getCompatibilityInfo()` and finally `json_decode(file_get_contents(...))`. Consequently, in a screen like "Admin => System => Extensions", it's plausible that `json_decode(file_get_contents(...))` would be executed 10-100 times (depending on how many extension are local and how many are in the public directory)?

Seems like a valid case for `Civi::$statics`?